apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/MongoDBMethodInterceptor.java

```diff
@@ -201,7 +201,13 @@ private String limitFilter(String filter) {
     public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
         Cluster cluster = (Cluster)allArguments[0];
         StringBuilder peers = new StringBuilder();
-        for (ServerDescription description : cluster.getDescription().getServerDescriptions()) {
+        Collection<ServerDescription> descriptions;
+        try {
+            descriptions = cluster.getDescription().getServerDescriptions();
+        } catch (Throwable t) {
+            descriptions = cluster.getDescription().getAll();
```

Comment:
Please use witness class to identify versions, using exception is very bad for performance.

Reply:
Thanks for your feeedback @wu-sheng 
I think I need to refer to some current code about witness class.
Any example code for using witness class?

Reply:
Witness class is very easy. Take `MongoDBCollectionInstrumentation` as example. 
Witness class is using a special class name in this version to tell agent core, only active the instrumentation when it exists.

Reply:
Thanks a lot :+1: 
I will try to use witeness class.